core/tests/test_utils.h

```diff
@@ -0,0 +1,290 @@
+        TEST_CHECK(utils_axis_index_to_mask_value(AxisIndexY) == AxesMaskY);
+        TEST_CHECK(utils_axis_index_to_mask_value(AxisIndexNY) == AxesMaskNY);
+        TEST_CHECK(utils_axis_index_to_mask_value(AxisIndexZ) == AxesMaskZ);
+        TEST_CHECK(utils_axis_index_to_mask_value(AxisIndexNZ) == AxesMaskNZ);
+    }
+}
+
+void test_string_new_join(void) {
+
+    char *str;
+    char *verif;
+
+    str = _string_new_join(1, "Hello");
+    verif = "Hello";
+    TEST_CHECK(strcmp(str, verif) == 0);
+    free(str);
+
+    str = _string_new_join(1, "123890 _&@/()[]{}");
+    verif = "123890 _&@/()[]{}";
+    TEST_CHECK(strcmp(str, verif) == 0);
+    free(str);
+
+    str = _string_new_join(5, "Hello", "World", "This", "Is", "Moon");
+    verif = "HelloWorldThisIsMoon";
+    TEST_CHECK(strcmp(str, verif) == 0);
+    free(str);
+}
+
+void test_string_new_copy(void) {
+
+    char *str;
+
+    str = string_new_copy("Hello");
+    TEST_CHECK(strcmp(str, "Hello") == 0);
+    free(str);
+
+    str = string_new_copy("01234 _&@/()[]{}");
+    TEST_CHECK(strcmp(str, "01234 _&@/()[]{}") == 0);
+    free(str);
+}
+
+void test_string_new_substring(void) {
+
+    char *start = "Hello";
+    char *end = "Hello World This Is Moon";
+    char *res;
+
+    res = string_new_substring(start, end);
+    TEST_CHECK(strcmp(res, start) == 0);
+    free(res);
+
+    start = "Hello Wolrd";
+    res = string_new_substring(start, end);
+    TEST_CHECK(strcmp(res, start) == 0);
+    free(res);
+
+    start = "Moon";
+    res = string_new_substring(start, end);
+    TEST_CHECK(strcmp(res, start) == 0);
+    free(res);
+}
+
+void test_string_new_copy_with_limit(void){
+
+    char *str = "This is a sentance";
+    char *res;
+
+    res = string_new_copy_with_limit(str, 9);
+    TEST_CHECK(strcmp(res, "This is a") == 0);
+    free(res);
+
+    res = string_new_copy_with_limit(str, 999);
+    TEST_CHECK(strcmp(res, "This is a sentance") == 0);
+    free(res);
+
+    res = string_new_copy_with_limit(str, 0);
+    TEST_CHECK(strcmp(res, "") == 0);
+    free(res);
+}
+
+void test_stringArray_new(void){
+
+    stringArray_t *arr = stringArray_new();
+    int lenght = stringArray_length(arr);
+    char *check = stringArray_get(arr, 0);
+    TEST_CHECK(check == NULL);
+    TEST_CHECK(lenght == 0);
+    stringArray_free(arr);
+}
+
+void test_stringArray_n_append(void){
+
+    bool verif;
+    char *str;
+    stringArray_t *arr = stringArray_new();
+
+    str = "Hello";
+    verif = stringArray_n_append(arr, str, 5);
+    TEST_CHECK(verif == true);
+    int lenght = stringArray_length(arr);
+    TEST_CHECK(lenght == 1);
+
+    str = "01234 _&@/()[]{}";
+    verif = stringArray_n_append(arr, str, 16);
+    TEST_CHECK(verif == true);
+    lenght = stringArray_length(arr);
+    TEST_CHECK(lenght == 2);
+
+    char *check = stringArray_get(arr, 0);
+    TEST_CHECK(strcmp(check, "Hello") == 0);
+    check = stringArray_get(arr, 1);
+    TEST_CHECK(strcmp(check, "01234 _&@/()[]{}") == 0);
```

Comment:
also, maybe we should test with a length smaller than the string ... let's talk about that this week.